src/simplification/SimplifyRequired.ts

```diff
@@ -0,0 +1,42 @@
+import { Schema } from 'models/Schema';
+
+type Output = string[] | undefined;
+
+/**
+ * Find the required array for a simplified version of a schema
+ * 
+ * @param schema to find the simplified required array for
+ * @param seenSchemas already seen schemas, this is to avoid circular schemas
+ */
+export default function simplifyRequired(schema: Schema | boolean, seenSchemas: Set<any> = new Set()): Output {
```

Comment:
Just gonna mention it, that others might find it weird that we all of the sudden use a `Set` over map like the other simplification rules, also in regard to you returning undefined instead of the previous already simplified property. Not saying one is better then the other, but should this be the same across the board? 🤔

Reply:
I don't need to return previous simplified property, because I only return array of values, not reference, due to this fact I changed Map to Set.

Reply:
I added comment why I don't need Map :P

Reply:
That is what I mean 😄 If someone else then us wants contribute to the simplifier code and looks through the separate functions then all the others does one behavior and this one does it differently. Question is do we want the same behavior across the board or do we not care about that? 🤔

Reply:
I think we should go with use case behaviour. In one place Set will be better, on another Map.

Reply:
Alright 👍